mmseg/apis/train.py

```diff
@@ -152,7 +152,7 @@ def train_segmentor(model,
 
     # register eval hooks
     if validate:
-        val_dataset = build_dataset(cfg.data.val, dict(test_mode=True))
+        val_dataset = build_dataset(cfg.data.test, dict(test_mode=True))
```

Comment:
I'm sorry but I didn't understand: you meant that `cfg.data.test` is the dataset cfg for test during training?

Reply:
sorry, I wrote it wrongly

Reply:
I see. For me the problem arises from the fact that during training the training and validation pipelines are interwined. If the two pipelines were separated (as I suggested [here](https://github.com/open-mmlab/mmsegmentation/pull/1759), then it would make sense to use the test pipeline also during the test phase while training.

Reply:
Thanks for the explanation, now I understand your concern.

I should make 3 kinds of dataset configuration, training, testing, and validation clearer.

Here is the configuration of three datasets and their augmentation pipelines.
https://github.com/open-mmlab/mmsegmentation/blob/13d4c3997c4ab3d5a56d355d09a712ada6fe0489/configs/_base_/datasets/ade20k.py#L34-L54

`train` and its pipeline are for model training, obviously, and `test` and its pipeline are for model testing to evaluate model performance with a specific metric like mIoU.

`val` and its pipeline, which are the same as `test` (but for more flexibility, we use two keys `test` and `val`), are also for evaluating model performance during training.

In addition, there is still a validation workflow that works for loss calculation of validation samples rather than evaluation metrics, so we use the validation dataset and training pipeline for this loop.